detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/BracesOnWhenStatements.kt

```diff
@@ -0,0 +1,254 @@
+ *  when (a) {
+ *      1 -> { f1() }
+ *      2 -> f2()
+ *  }
+ *  // multiLine = 'never'
+ *  when (a) {
+ *      1 -> {
+ *          f1()
+ *      }
+ *      2 -> f2()
+ *  }
+ *  // singleLine = 'necessary'
+ *  when (a) {
+ *      1 -> { f1() }
+ *      2 -> f2()
+ *  }
+ *  // multiLine = 'necessary'
+ *  when (a) {
+ *      1 -> { f1() }
+ *      2 -> { f2(); f3() }
+ *  }
+ *
+ *  // singleLine = 'consistent'
+ *  when (a) {
+ *      1 -> { f1() }
+ *      2 -> f2()
+ *  }
+ *  // multiLine = 'consistent'
+ *  when (a) {
+ *      1 ->
+ *          f1()
+ *      2 -> {
+ *          f2()
+ *          f3()
+ *      }
+ *  }
+ *
+ *  // singleLine = 'always'
+ *  when (a) {
+ *      1 -> { f1() }
+ *      2 -> f2()
+ *  }
+ *  // multiLine = 'always'
+ *  when (a) {
+ *      1 ->
+ *          f1()
+ *      2 -> {
+ *          f2()
+ *          f3()
+ *      }
+ *  }
+ *
+ *  </noncompliant>
+ *
+ *  <compliant>
+ *  // singleLine = 'never'
+ *  when (a) {
+ *      1 -> f1()
+ *      2 -> f2()
+ *  }
+ *  // multiLine = 'never'
+ *  when (a) {
+ *      1 ->
+ *          f1()
+ *      2 -> f2()
+ *  }
+ *  // singleLine = 'necessary'
+ *  when (a) {
+ *      1 -> f1()
+ *      2 -> f2()
+ *  }
+ *  // multiLine = 'necessary'
+ *  when (a) {
+ *      1 ->
+ *          f1()
+ *      2 -> {
+ *          f2()
+ *          f3()
+ *      }
+ *  }
+ *
+ *  // singleLine = 'consistent'
+ *  when (a) {
+ *      1 -> { f1() }
+ *      2 -> { f2() }
+ *  }
+ *  when (a) {
+ *      1 -> f1()
+ *      2 -> f2()
+ *  }
+ *  // multiLine = 'consistent'
+ *  when (a) {
+ *      1 -> {
+ *          f1()
+ *      }
+ *      2 -> {
+ *          f2()
+ *          f3()
+ *      }
+ *  }
+ *
+ *  // singleLine = 'always'
+ *  when (a) {
+ *      1 -> { f1() }
+ *      2 -> { f2() }
+ *  }
+ *  // multiLine = 'always'
+ *  when (a) {
+ *      1 -> {
+ *          f1()
+ *      }
+ *      2 -> {
+ *          f2()
+ *          f3()
+ *      }
+ *  }
+ *
+ *  </compliant>
+ */
+class BracesOnWhenStatements(config: Config = Config.empty) : Rule(config) {
+    override val issue = Issue(
+        javaClass.simpleName,
+        Severity.Style,
+        "Braces do not comply with the specified policy",
+        Debt.FIVE_MINS
+    )
+
+    @Configuration("single-line braces policy")
+    private val singleLine: BracePolicy by config("consistent") { BracePolicy.getValue(it) }
+
+    @Configuration("multi-line braces policy")
+    private val multiLine: BracePolicy by config("consistent") { BracePolicy.getValue(it) }
+
+    override fun visitWhenExpression(expression: KtWhenExpression) {
+        super.visitWhenExpression(expression)
+
+        val branches: List<KtExpression> = walk(expression)
+        validate(branches, policy(expression))
+    }
+
+    private fun walk(expression: KtWhenExpression): List<KtExpression> {
+        return expression.entries.mapNotNull { it.expression }
+    }
+
+    private fun validate(list: List<KtExpression>, policy: BracePolicy) {
```

Comment:
```suggestion
    private fun validate(branches: List<KtExpression>, policy: BracePolicy) {
```
should improve readability of the function.

(maybe rename in `if` rule too?)